php/class-coauthors-guest-authors.php

```diff
@@ -463,36 +463,70 @@ function view_guest_authors_list() {
 				wp_die( esc_html( sprintf( __( "%s can't be deleted because it doesn't exist.", 'co-authors-plus' ), $this->labels['singular'] ) ) );
 			}
 
+			// get post count
+			global $coauthors_plus;
+			$term = $coauthors_plus->get_author_term( $guest_author );
+			if ( $term ) {
+				$count = $term->count;
+			} else {
+				$count = 0;
+			}
+
 			echo '<div class="wrap">';
 			echo '<div class="icon32" id="icon-users"><br/></div>';
 			echo '<h2>' . esc_html( sprintf( __( 'Delete %s', 'co-authors-plus ' ), $this->labels['plural'] ) ) . '</h2>';
 			echo '<p>' . esc_html(  sprintf( __( 'You have specified this %s for deletion:', 'co-authors-plus' ), strtolower( $this->labels['singular'] ) ) ) . '</p>';
 			echo '<p>#' . esc_html( $guest_author->ID . ': ' . $guest_author->display_name ) . '</p>';
-			echo '<p>' . esc_html(  sprintf( __( 'What should be done with posts assigned to this %s?', 'co-authors-plus' ), strtolower( $this->labels['singular'] ) ) ) . '</p>';
-			echo '<p class="description">' . esc_html( sprintf( __( "Note: If you'd like to delete the %s and all of their posts, you should delete their posts first and then come back to delete the %s.", 'co-authors-plus' ), strtolower( $this->labels['singular'] ), strtolower( $this->labels['singular'] ) ) ) . '</p>';
+			// display wording differently per post count
+			if ( 0 === $count ) {
+				$post_count_message = '<p>' . esc_html(  sprintf( __( 'There are no posts associated with this guest author.', 'co-authors-plus' ), strtolower( $this->labels['singular'] ) ) ) . '</p>';
+			}
+			else {
+				$note = '<p class="description">' . esc_html( sprintf( __( "Note: If you'd like to delete the %s and all of their posts, you should delete their posts first and then come back to delete the %s.", 'co-authors-plus' ), strtolower( $this->labels['singular'] ), strtolower( $this->labels['singular'] ) ) ) . '</p>';
+				if ( 1 === $count ) {
+					$post_count_message = '<p>' . esc_html(  sprintf( __( 'There is %d post associated with this guest author. What should be done with the post assigned to this %s?', 'co-authors-plus' ), $count, strtolower( $this->labels['singular'] ) ) ) . '</p>';
+				}
+				else {
+					$post_count_message = '<p>' . esc_html(  sprintf( __( 'There are %d posts associated with this guest author. What should be done with the posts assigned to this %s?', 'co-authors-plus' ), $count, strtolower( $this->labels['singular'] ) ) ) . '</p>';
+				}
+				$post_count_message .= $note;
+			}
+			echo $post_count_message;
```

Comment:
It's always best to [late escape](https://vip.wordpress.com/documentation/vip/best-practices/security/validating-sanitizing-escaping/#always-escape-late) to avoid future changes making the output unsafe. This also simplifies the above code by removing the need for multiple `esc_html()` calls.